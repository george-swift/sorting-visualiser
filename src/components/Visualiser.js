/* eslint-disable react/no-array-index-key */
import PropTypes from 'prop-types';
import { PRIMARY_COLOR } from '../utils';

const Visualiser = ({ array, reference }) => (
  <div className="visualiser" ref={reference}>
    {array.map((value, index) => (
      <div
        className="vsx-bar"
        key={index}
        style={{
          backgroundColor: PRIMARY_COLOR,
          height: `${value}px`,
        }}
      />
    ))}
  </div>
);

Visualiser.propTypes = {
  array: PropTypes.arrayOf(PropTypes.number).isRequired,
  reference: PropTypes.objectOf(PropTypes.object).isRequired,
};

export default Visualiser;
