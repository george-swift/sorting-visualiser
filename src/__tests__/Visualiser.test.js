import React from 'react';
import renderer from 'react-test-renderer';
import Visualiser from '../components/Visualiser';

describe('Snapshot of the Visualiser component', () => {
  const mockRef = React.createRef();
  const array = Array(5).fill(6);

  it('should render correctly', () => {
    const tree = renderer.create(
      <Visualiser array={array} reference={mockRef} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
