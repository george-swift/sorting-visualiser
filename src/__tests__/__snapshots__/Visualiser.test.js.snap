// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Visualiser component should render correctly 1`] = `
<div
  className="visualiser"
>
  <div
    className="vsx-bar"
    style={
      Object {
        "backgroundColor": "turquoise",
        "height": "6px",
      }
    }
  />
  <div
    className="vsx-bar"
    style={
      Object {
        "backgroundColor": "turquoise",
        "height": "6px",
      }
    }
  />
  <div
    className="vsx-bar"
    style={
      Object {
        "backgroundColor": "turquoise",
        "height": "6px",
      }
    }
  />
  <div
    className="vsx-bar"
    style={
      Object {
        "backgroundColor": "turquoise",
        "height": "6px",
      }
    }
  />
  <div
    className="vsx-bar"
    style={
      Object {
        "backgroundColor": "turquoise",
        "height": "6px",
      }
    }
  />
</div>
`;
